docs/src/ref_impl/installer.md

```diff
@@ -5,67 +5,56 @@
-          sync
-          echo "Flushing finished successfully!"
-          echo "Now you can detach installation device and reboot to Ghaf."
-      else
-          echo "Some error occured during flushing process, exit code: $?."
-          exit
-      fi
-    '';
-  };
-}
-```
+Ghaf has a NixOS-like non-interactive, declarative installer instead of a
+conventional (imperative, non-declarative) installer as in most conventional
+Linux distributions. This is possible with the [Ghaf as
+Library](https://github.com/tiiuae/ghaf/pull/ghaf-based-project.md) approach:
+rather than clicking similar options during installation, you can configure the
+system once and deploy this configuration to the desired machines.
+
+
+To implement the pre-configured setting up approach, we used the
+[nixos-anywhere](https://github.com/nix-community/nixos-anywhere) tool.
+
+There are two separate ways to use it:
+* manually
+* with the Ghaf installation wizard.
+> NOTE: The Ghaf installation wizard is currently under development and cannot be used to create the required system configuration.
 
-After that you can build an installer image using this command:
+## Manual Installation
+
+To install Ghaf manually:
+
+1. Create your own flake using the Ghaf template:
 
 ```sh
-nix build .#nixosConfigurations.<CONFIGURATION>.config.system.build.installer
+nix flake init -t github:tiiuae/ghaf#target-x86_64-generic
 ```
 
-## Adding Installer Modules
+2. Edit it according to your preferences.
+
+3. Set the value of `ghaf.installer.sshKeys` to get an installer image.
 
-To add an installer module, replace the corresponding placeholders with your code and add this to your configuraiton:
+4. Build the installer image:
 
-```nix
-ghaf.installer.installerModules.<MODULE_NAME> = {
-  requestCode = ''
-    # Your request code written in Bash
-  '';
-  providedVariables = {
-    # Notice the dollar sign before the actual variable name in Bash.
-    <VARIABLE_NAME> = "$<VARIABLE_NAME>";
-  };
-};
+```sh
+nix build .#nixosConfigurations.PROJ_NAME-ghaf-debug.config.system.build.installer
 ```
 
-## Built-in Installer Modules
+5. Flash the installer image to your device:
+
+```sh
+sudo dd if=./result/iso/nixos-...-linux.iso of=/dev/YOUR_DEVICE conv=sync && sync
+```
 
-Provided variables show variable names in Nix. For actual names of variables in Bash, see the sources of the module.
+6. Run the image on the device.
 
-### flushImage
+7. Connect the device to the network using `wifi-connector`.
 
-Provided variables:
+8. Get a block device name and put it in a disk configuration using the `lsblk` command.
 
-- deviceName: name of the device on which image should be flushed (e.g. "sda", "nvme0n1")
+8. Install the NixOS configuration to the target device using the command:
 
+```sh
+nix run github:numtide/nixos-anywhere -- --flake .#CONFIGURATION_NAME root@IP_ADDRESS
+```
```

Comment:
After all the installer seems to work. After installation target device booted ghaf from the internal NVMe SSD.

![IMG_2225](https://github.com/tiiuae/ghaf/assets/129752236/1863d984-668d-413f-86fd-c328103bd537)